### ISSUE TITLE
Training batch size and performance on LibriMix

### ISSUE THREAD

**Issue by Galaxy-ZRX** [JusperLee/S4M#7]
Hi Jusper, thank you for your amazing job. May I ask which batch size did you use during the training of S4M? I didn't find it in the paper, and for TDANet you used batch size = 1. I used batch size = 8 with lr unchanged (1e-3), trained for 150 epochs (using the audio_train in TDANet codes), but I only got a result around 14.3/14.8, which is a bit far from the reported S4M-tiny (16.2, 16.6). Below attached my configs. Could you please check if I made something wrong which led to the performance drop? btw I have followed another issue to modify the enc_kernel_size from 21 to 4, to make it consistent with default S4M-tiny. 

<img width="256" height="196" alt="Image" src="https://github.com/user-attachments/assets/e1878f9c-a2f7-4a90-b949-6f703a72a53e" />

```
audionet:
  audionet_config:
    num_sources: 2
  audionet_name: S4M
datamodule:
  data_config:
    batch_size: 8
    n_src: 2
    normalize_audio: false
    num_workers: 8
    persistent_workers: false
    pin_memory: true
    sample_rate: 8000
    segment: 3.0
    test_dir: DataPreProcess/Libri2Mix/test
    train_dir: DataPreProcess/Libri2Mix/train-100
    valid_dir: DataPreProcess/Libri2Mix/dev
  data_name: Libri2MixDataModule
exp:
  exp_name: S4M
loss:
  train:
    config:
      pit_from: pw_mtx
      threshold_byloss: true
    loss_func: PITLossWrapper
    sdr_type: pairwise_neg_snr
  val:
    config:
      pit_from: pw_mtx
      threshold_byloss: false
    loss_func: PITLossWrapper
    sdr_type: pairwise_neg_sisdr
main_args:
  exp_dir: --removed--
optimizer:
  lr: 0.001
  optim_name: adam
  weight_decay: 0
options:
  conf_dir: configs/s4m.yml
  help: null
scheduler:
  sche_config:
    factor: 0.5
    patience: 15
  sche_name: ReduceLROnPlateau
training:
  SpeedAug: false
  early_stop:
    mode: min
    monitor: val_loss/dataloader_idx_0
    patience: 30
    verbose: true
  epochs: 200
  gpus:
  - 0
  system: AudioLightningModule
```

Thank you again for your work and look forward to your reply :-)

**Comment by JusperLee** [comment]
> lr unchanged (1e-3), trained for 150 epochs (using the audio_train in TDANet codes), but I only got a result around 14.3/14.8

I think this is normal because speech separation models often require a longer convergence process. Generally speaking, the learning rate needs to decrease three to four times before it fully converges.

**Comment by Galaxy-ZRX** [comment]
Hi Jusper, thank you very much for your reply. Just to confirm that should I train the S4M for 200 epochs in total as described in the paper? I have finished another 200-epochs training with batch size adjusted to 4, and still got only 14.48 / 14.96. Here also attached my logs. Thank you for your help again.

<img width="553" height="341" alt="Image" src="https://github.com/user-attachments/assets/938bee35-fb0c-40b6-9544-af285fae6ae6" />

<img width="378" height="259" alt="Image" src="https://github.com/user-attachments/assets/71406264-953b-4c8a-a8f3-d1a393d1bdfd" />

**Comment by Galaxy-ZRX** [comment]
I noticed that it's partly because I didn't use audio_test for testing, instead I used simple codes to load the model and calculate the metrics. After updating this by using audio_test, I got the below results, sdr_i: 15.946 si-snr_i: 15.484, which is better but still not reached the paper's results. May I ask if such results are acceptable, e.g., maybe due to random seeds?

**Comment by JusperLee** [comment]
Can I take a look at the learning rate? Generally, you need the learning rate to decrease several times before it converges. It looks like yours hasn’t converged yet.

**Comment by Galaxy-ZRX** [comment]
Yes the lr is always 1e-3 without decrease. I didn't change anything from the conf related to lrdecay as below:
`scheduler:
  sche_config:
    factor: 0.5
    patience: 15
  sche_name: ReduceLROnPlateau`

but since the best performance is kept updating in every 15 epochs, the lr didnt get a chance to decrease I guess? I remember for he first time I trained the model, the lr decreased once from 1e-3 to 5e-4, but this time it didn't. Should I modify the patience from 15 to lower numbers to make sure the lr can decrease?

**Comment by Galaxy-ZRX** [comment]
Hi Jusper, may I ask if it's possible to get your pretrained model (.pth file) on libri2mix? I am also testing with your TDANet code, and both of them cannot reach the paper results (~2% lower), it would be really helpful if I can compare with the trained model of yours. thank you!

**Comment by JusperLee** [comment]
This training was done by the first author; I just asked him for the code. I can spend a week reproducing the results, and then I’ll update the codebase.

**Comment by Galaxy-ZRX** [comment]
Thank you! I have trained TDANet for 500 epochs, and achieved sdri: 15.42, si-snri: 14.96, using the default config file. I will not open another issue in TDANet repo and keep the discussions here if that's ok.

**Comment by Galaxy-ZRX** [comment]
Hi Jusper, just to ask again if currently there is TDANet pretrained model (.ckpt or .pth files) available? In its repo, you provided the model trained on LRS2 in a closed issue, and I was wondering if the model for LibriMix is also available. Thank you so much for your help!
